extensions/wikia/Discussions/maintenance/ForumDumper.php

```diff
@@ -0,0 +1,246 @@
+						"last_child_comment_id" => $row->last_child_comment_id,
+						"archived_ind" => $row->archived ?: 0,
+						"deleted_ind" => $row->deleted ?: 0,
+						"removed_ind" => $row->removed ?: 0,
+						"locked_ind" => $row->locked ?: 0,
+						"protected_ind" => $row->protected ?: 0,
+						"sticky_ind" => $row->sticky ?: 0,
+						"first_revision_id" => $row->first_rev_id,
+						"last_revision_id" => $row->last_rev_id,
+						"comment_timestamp" => $row->last_touched
+					] );
+			} );
+
+		return $this->pages;
+	}
+
+	public function getRevisions() {
+		if ( !empty( $this->revisions ) ) {
+			return $this->revisions;
+		}
+		
+		$pageIds = array_keys( $this->getPages() );
+
+		$dumper = $this;
+		$dbh = wfGetDB( DB_SLAVE );
+		( new \WikiaSQL() )
+			->SELECT_ALL()
+			->FROM( self::TABLE_REVISION )
+			->JOIN( self::TABLE_TEXT )->ON( 'rev_text_id', 'old_id' )
+			->WHERE( 'rev_page' )->IN( $pageIds )
+			->runLoop( $dbh, function ( &$revisions, $row ) use ( $dumper ) {
+				$textId = $row->old_text;
+				list( $parsedText, $plainText, $title ) = $dumper->getTextAndTitle( $textId );
+
+				$pages = $dumper->getPages();
+				$curPage = $pages[$row->rev_page];
+				
+				$dumper->addRevision( [
+					"revision_id" => $row->rev_id,
+					"page_id" => $row->rev_page,
+					"page_namespace" => $curPage['namespace'],
+					"title" => $title,
+					"user_type" => $dumper->getContributorType( $row ),
+					"user_identifier" => $row->rev_user,
+					"timestamp" => $row->rev_timestamp,
+					"is_minor_edit" => $row->rev_minor_edit,
+					"is_deleted" => $row->rev_deleted,
+					"length" => $row->rev_len,
+					"parent_id" => $row->rev_parent_id,
+					"text_flags" => $row->old_flags,
+					"comment" => $row->rev_comment,
+					"raw_content" => $plainText,
+				    "content" => $parsedText
+				] );
+			} );
+
+		return $this->revisions;
+	}
+
+	public function getTextAndTitle( $textId ) {
+		$rawText = $this->getRawText( $textId );
+
+		// The title is included within the text as an ac_metadata tag
+		$title = $this->getTitle( $rawText );
+
+		// Parse the wiki text
+		$parsedText = $this->getParsedText( $rawText );
+
+		// Get a plain text version as well
+		$plainText = strip_tags( $parsedText );
+
+		return [ $parsedText, $plainText, $title ];
+	}
+
+	private function getRawText( $textId ) {
+		$text = \ExternalStore::fetchFromURL( $textId );
+		return gzinflate( $text );
+	}
+
+	private function getParsedText( $wikiText ) {
+		$dummyTitle = \Title::newFromText( "Dummy" );
+		$parserOptions = new \ParserOptions();
+		$parserOptions->setEditSection( false );
+		$parserOut = \F::app()->wg->Parser->parse( $wikiText, $dummyTitle, $parserOptions );
+		return $parserOut->getText();
+	}
+
+	private function getTitle( $text ) {
+		$title = '';
+
+		if ( preg_match( self::REGEXP_MATCH_TITLE, $text, $matches ) ) {
+			$title = $matches[1];
+		}
+
+		return $title;
+	}
+
+	public function getContributorType( $row ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)